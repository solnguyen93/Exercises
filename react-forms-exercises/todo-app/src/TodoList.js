import React, { useState } from 'react';
import Todo from './Todo.js';
import NewTodoForm from './NewTodoForm';
import { v4 as uuid } from 'uuid';

function TodoList() {
    const INITIAL_STATE = [];

    const [Todos, setTodos] = useState(INITIAL_STATE);

    const addTodo = (newTodo) => {
        setTodos((Todos) => [...Todos, { ...newTodo, id: uuid() }]);
    };

    const handleDelete = (todoId) => {
        setTodos((Todos) => Todos.filter((todo) => todo.id !== todoId));
    };
    return (
        <div className="TodoList">
            <h3>Todo List</h3>
            <NewTodoForm addTodo={addTodo} />
            <div className="TodoList-Todos">
                {Todos.map(({ id, task }) => (
                    <Todo key={id} task={task} onDelete={() => handleDelete(id)} />
                ))}
            </div>
        </div>
    );
}

export default TodoList;
