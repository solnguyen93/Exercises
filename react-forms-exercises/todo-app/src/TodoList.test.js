import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

it('renders', function () {
    render(<TodoList />);
});

it('matches snapshot', function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it('should add new todo', function () {
    const { queryByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText('Task');
    const addBtn = queryByText('Add Task');

    expect(queryByText('X')).not.toBeInTheDocument();

    fireEvent.change(input, { target: { value: 'eat, sleep, study' } });
    fireEvent.click(addBtn);

    expect(queryByText('eat, sleep, study')).toBeInTheDocument();
});

it('should delete todo', function () {
    const { queryByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText('Task');
    const addBtn = queryByText('Add Task');

    expect(queryByText('X')).not.toBeInTheDocument();

    fireEvent.change(input, { target: { value: 'eat, sleep, study' } });
    fireEvent.click(addBtn);

    expect(queryByText('eat, sleep, study')).toBeInTheDocument();

    const deleteBtn = queryByText('X');
    fireEvent.click(deleteBtn);
    expect(queryByText('X')).not.toBeInTheDocument();
    expect(queryByText('eat, sleep, study')).not.toBeInTheDocument();
});
