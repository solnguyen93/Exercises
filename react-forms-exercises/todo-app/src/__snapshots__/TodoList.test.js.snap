// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h3>
      Todo List
    </h3>
    <form>
      <label
        for="task"
      >
        Task
      </label>
      <input
        id="task"
        name="task"
        placeholder="task"
        type="text"
        value=""
      />
      <button>
        Add Task
      </button>
    </form>
    <div
      class="TodoList-Todos"
    />
  </div>
</DocumentFragment>
`;
